pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -306,14 +315,41 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 			return nil, httperror.WrapAPIError(err, httperror.ServerError, "Error processing clusterTemplate answers")
 		}
 		keyParts := strings.Split(question.Variable, ".")
-		values.PutValue(dataFromTemplate, val, keyParts...)
+		if strings.HasPrefix(question.Variable, "rancherKubernetesEngineConfig.privateRegistries") {
+			// for example: question.Variable = rancherKubernetesEngineConfig.privateRegistries[0].url
+			index, err := getIndexFromQuestion(question.Variable)
+			if err != nil {
+				return nil, httperror.WrapAPIError(err, httperror.ServerError, "Error processing clusterTemplate answers to private registry")
+			}
+			key := keyParts[len(keyParts)-1]
+			registry, ok := registryMap[index]
+			if ok {
+				registry[key] = val
+				registryMap[index] = registry
```

Comment:
Can these be combined? Then you wouldn't need the `registry` variable. This would save an extra assignment, but I guess is mostly a style thing. So, consider it a nit.
```suggestion
				registryMap[index][key] = val
```

Reply:
fixed.